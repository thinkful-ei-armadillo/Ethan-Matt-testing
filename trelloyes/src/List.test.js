import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';

const cards = [
  {
    key     : 'a',
    title   : 'Card Title',
    content : 'Card Content',
  },
  {
    key     : 'b',
    title   : 'Card Title',
    content : 'Card Content',
  },
  {
    key     : 'c',
    title   : 'Card Title',
    content : 'Card Content',
  },
];

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<List header="Test Header" cards={cards} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
  .create(<List header="Test Header" cards={cards} />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
