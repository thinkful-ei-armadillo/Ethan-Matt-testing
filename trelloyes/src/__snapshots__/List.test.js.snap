// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Test Header
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        Card Title
      </h3>
      <p>
        Card Content
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Card Title
      </h3>
      <p>
        Card Content
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Card Title
      </h3>
      <p>
        Card Content
      </p>
    </div>
  </div>
</section>
`;
